server/src/services/emailService.ts

```
@@ -0,0 +1,107 @@
+import logger from './logger.js';
+
+export interface EmailService {
+  sendVerificationEmail: (params: { email: string; username: string; token: string }) => Promise<void>;
+  sendPasswordResetEmail: (params: { email: string; username: string; token: string }) => Promise<void>;
+}
+
+function buildVerificationEmail(params: { email: string; username: string; token: string }) {
+  const { email, token } = params;
+  // Sanitize username: remove control characters and newlines
+  const safeUsername = params.username
+    .replace(/[\r\n\t\x00-\x1F\x7F]/g, '')
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)